src/packet.rs

```diff
@@ -96,29 +112,84 @@ impl MspParser {
 			},
 
 			MspParserState::Header2 => {
-				if input == ('M' as u8) {
-					self.state = MspParserState::Direction;
-				} else {
-					self.reset();
-				}
+				self.packet_version = match input as char {
+					'M' => MSPVersion::V1,
+					'X' => MSPVersion::V2,
+					_=> {
+						self.reset();
+						return Ok(None);
+					},
+				};
+
+				self.state = MspParserState::Direction;
 			},
 
 			MspParserState::Direction => {
 				match input {
-					60 => self.packet_direction = MspPacketDirection::ToFlightController,
-					62 => self.packet_direction = MspPacketDirection::FromFlightController,
-					33 => self.packet_direction = MspPacketDirection::Unsupported,
+					60 => self.packet_direction = MspPacketDirection::ToFlightController, // '>'
+					62 => self.packet_direction = MspPacketDirection::FromFlightController, // '<'
+					33 => self.packet_direction = MspPacketDirection::Unsupported, // '!' error
 					_ => {
 						self.reset();
 						return Ok(None);
 					}
 				}
 
-				self.state = MspParserState::DataLength;
+				self.state = match self.packet_version {
+					MSPVersion::V1 => MspParserState::DataLength,
+					MSPVersion::V2 => MspParserState::FlagV2,
+				};
 			},
 
-			MspParserState::DataLength => {
+			MspParserState::FlagV2 => {
+				// uint8, flag, usage to be defined (set to zero)
+				self.state = MspParserState::CommandV2;
+				self.packet_data = Vec::with_capacity(2);
```

Comment:
This breaks the code for `no_std`. Consider static arrays internally and passing slices (`&[u8]`) for input. Vec requires a heap, which is not given in `no_std` without additional crates.

Reply:
`packet_data` is already `Vec<>`, so that means no_std is already broken.
also how do i go about validating no_std is working, i tried creating dummy project with `#![no_std]`, similar to https://github.com/hashmismatch/packed_struct.rs/tree/master/packed_struct_nostd_tests. but it just won't compile.

Reply:
i think we should fork this repo, and address no_std in separate PR